app/models/link.rb

```diff
@@ -1,7 +1,9 @@
 class Link < Datafile
   attr_accessor :day, :month, :year
-  
-  before_save :set_end_date
+
+  # Some legacy datafiles have invalid dates (e.g. 31/06/15).
+  # When this occurs the date attributes are empty, therefore we cannot invoke this call-back
+  before_save :set_end_date, unless: ->{ year.nil? }
```

Comment:
will this work with daily datasets, or datafiles without dates (which can be saved even if they don't have a year)

Reply:
not sure i understand why you're only setting end date if the year isn't nil. Rather than only setting it if the start date is nil. Can you explain why this is better?